core/domain/suggestion_registry.py

```diff
@@ -635,6 +635,179 @@ def pre_update_validate(self, change):
                 'question_dict')
 
 
+class BaseVoiceoverApplication(python_utils.OBJECT):
```

Comment:
I'm wondering why you need an inheritance structure here -- are you expecting that there would be more subclasses of BaseVoiceoverApplication?

Reply:
We are expecting to have a voiceover application for other entity in the future (Like Story, topic etc.)

Reply:
Won't all the different entities just have different target_type and target_ids? I don't see why we will need an entire new subclass for them.

Reply:
So from what I gather, all the data fields and functions will be the exact same? The only thing that is different is the target_type right? In which case, can't you just have multiple functions in the service layer to create applications with appropriate target_types... I doubt that we need the inheritance based classes for this scenario

Reply:
Thoughts here @DubeySandeep?

Reply:
Hmmm, ok let's keep the inheritance in that case.